addons/pos_sale/static/src/js/OrderManagementScreen/SaleOrderManagementScreen.js

```diff
@@ -293,7 +293,9 @@ odoo.define('pos_sale.SaleOrderManagementScreen', function (require) {
             sale_order[0].amount_unpaid = saleOrdersAmountUnpaid[sale_order[0].id];
 
             const sale_lines = await this._getSOLines(sale_order[0].order_line);
-            sale_order[0].order_line = sale_lines;
+            const program_by_id = this.env.pos.coupon_programs_by_id;
+            const promo_products_to_remove = Object.keys(program_by_id).flatMap(pId => program_by_id[pId].promo_code_usage === 'no_code_needed' ? program_by_id[pId].discount_line_product_id[0] : []);
```

Comment:
It works as it is but I think we do not need the ids of the programs so maybe do it like this ?
```suggestion
            const promo_products_to_remove = this.env.pos.coupon_programs.flatMap(program => program.promo_code_usage === 'no_code_needed' ? program.discount_line_product_id[0] : []);
```

Reply:
coupon_programs was empty, the program was in the `promo_programs`.